File: Python/codeup_1099_설실한 개미.py
arr = []

for i in range(10):
    arr.append([])
    for j in range(10):
        arr[i].append(0)

for i in range(10):
    num = input().split()
    for j in range(10):
        arr[i][j] = int(num[j])

a = 1
b = 1

for i in range(15):
    if arr[a][b] == 0:
        arr[a][b] = 9
        b += 1

    elif arr[a][b] == 1:
        a += 1
        b -= 1

        if arr[a][b] == 2:
            arr[a][b] = 9
            break

        arr[a][b] = 9
        b += 1

    elif arr[a][b] == 2:
        arr[a][b] = 9
        break

for i in range(10):
    for j in range(10):
        print(arr[i][j], end =' ')
    print()    

File: C++/codeup_1099_설실한 개미.cpp
#include<iostream>
using namespace std;

int main()
{
    int arr[10][10] = {};
    int a = 1;
    int b = 1;

    for(int i = 0; i < 10; i++)
        for(int j = 0; j < 10; j++)
            cin >> arr[i][j];

    for(int i = 0; i < 15; i++)
    {
        if(arr[a][b] == 0)
        {
            arr[a][b] = 9;
            b++;
        }

        else if(arr[a][b] == 1)
        {
            a++;
            b--;
            
            if(arr[a][b] == 2)
            {
                arr[a][b] = 9;
                break;
            }

            arr[a][b] = 9;
            b++; 
        }

        else if(arr[a][b] == 2)
        {
            arr[a][b] = 9;
            
            break;
        }
    }

    for(int i = 0; i < 10; i++)
    {
        for(int j = 0; j < 10; j++)
        {
             cout << arr[i][j] << " ";
        }
        cout << endl;
    }

    return 0;
}

